service/service.go

```diff
@@ -29,13 +34,23 @@ type Service struct {
 	// are correctly handled.
 	*onet.ServiceProcessor
 	path string
-	// Count holds the number of calls to 'ClockRequest'
+
+	storage *storage
+}
+
+// storageID reflects the data we're storing - we could store more
+// than one structure.
+const storageID = "main"
+
+// storage is used to save our data.
+type storage struct {
 	Count int
+	sync.Mutex
```

Comment:
I deleted the mutex and on my machine I have no race conditions. Do you have it on your machine ?
As an example, I think it'd be best if we try to avoid marshalling and unmarshalling strange things such as a mutex. We don't know what's happenning then.